contrib/babelfishpg_tsql/src/pltsql_utils.c

```diff
@@ -1791,3 +1792,125 @@ List
 
 	return stmt_list;
 }
+
+/*
+ * get_fulltext_index_name
+ * Get the fulltext index name of a relation specified by OID
+ */
+char *
+get_fulltext_index_name(Oid relid, const char *table_name)
+{
+    Relation        relation = RelationIdGetRelation(relid);
+    List            *indexoidlist = RelationGetIndexList(relation);
+	ListCell		*cell;
+    char            *ft_index_name = NULL;
+	char			*table_name_cpy = palloc(strlen(table_name) + 1);
+    char            *temp_ft_index_name;
+
+	strcpy(table_name_cpy, table_name);
+	temp_ft_index_name = construct_unique_index_name("ft_index", table_name_cpy);
+    foreach(cell, indexoidlist)
+    {
+        Oid indexOid = lfirst_oid(cell);
+        ft_index_name = get_rel_name(indexOid);
+
+        if (strcmp(ft_index_name, temp_ft_index_name) == 0)
+            break;
+
+        ft_index_name = NULL;
+    }
+
+    RelationClose(relation);
+    list_free(indexoidlist);
+	pfree(table_name_cpy);
+    return ft_index_name;
+}
+
+/*
+ * is_unique_index
+ * Check if given index is unique index of a relation specified by OID
+ */
+bool
+is_unique_index(Oid relid, const char *index_name)
+{
+    Relation        relation = RelationIdGetRelation(relid);
+    List            *indexoidlist = RelationGetIndexList(relation);
+	ListCell		*cell;
+    bool            is_unique = false;
+
+    foreach(cell, indexoidlist)
+    {
+        Oid indexOid = lfirst_oid(cell);
+        char *name = get_rel_name(indexOid);
+
+        if (strcmp(name, index_name) == 0)
+        {
+            HeapTuple       indexTuple = SearchSysCache1(INDEXRELID, ObjectIdGetDatum(indexOid));
+            Form_pg_index   indexForm = (Form_pg_index) GETSTRUCT(indexTuple);
+
+            if (indexForm->indisunique && indexForm->indrelid == relid)
```

Comment:
per https://github.com/MicrosoftDocs/sql-docs/blob/live/docs/t-sql/statements/create-fulltext-index-transact-sql.md#key-index-index_name, in addition to being unique, the index, must also be non-nullable and single-key. I am not sure of the pg_index tuple has metadata for all of these conditions, but if it does we might as well include them in this check as well.